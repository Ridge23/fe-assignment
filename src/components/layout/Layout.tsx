import React from 'react';

import SideMenu from 'components/sidemenu/SideMenu';
import TopBar from 'components/topbar/TopBar';
import Footer from 'components/footer/Footer';

import './sass/layout.scss';

interface ILayout {
    title: string;
    children: JSX.Element | JSX.Element[] | string;
}

export default function Layout({ title, children }: ILayout): JSX.Element {
    return (
        <div className="layout">
            <div className="layout__sidebar">
                <SideMenu />
            </div>
            <div className="layout__content">
                <div className="layout__content-header">
                    <TopBar />
                </div>
                <div className="layout__content-container">
                    <>
                        <h1 className="layout__content-title">{title}</h1>
                        {children}
                    </>
                </div>
                <div className="layout__content-footer">
                    <Footer />
                </div>
            </div>
        </div>
    )
}