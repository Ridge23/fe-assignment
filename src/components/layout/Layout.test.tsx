import React from 'react';
import { shallow } from 'enzyme';

import Layout from './Layout';

describe('Layout', () => {
    it('should render correctly', () => {
        const component = shallow(<Layout title='test_title'>test_content</Layout>);
        const heading = component.find('.layout__content-title');
        const contentContainer = component.find('.layout-content-container');

        expect(heading.text()).toEqual('test_title');
        expect(contentContainer.text()).toEqual('test_content');

        expect(component).toMatchSnapshot();
    });
});