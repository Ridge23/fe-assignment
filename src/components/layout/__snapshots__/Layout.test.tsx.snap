// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<div
  className="layout"
>
  <div
    className="layout__sidebar"
  >
    <SideMenu />
  </div>
  <div
    className="layout__content"
  >
    <div
      className="layout__content-header"
    >
      <TopBar />
    </div>
    <div
      className="layout__content-container"
    >
      <h1
        className="layout__content-title"
      >
        test_title
      </h1>
      <div
        className="layout-content-container"
      >
        test_content
      </div>
    </div>
    <div
      className="layout__content-footer"
    >
      <Footer />
    </div>
  </div>
</div>
`;
